src/main/java/org/aksw/simba/lemming/metrics/single/StdDevVertexDegree.java

```diff
@@ -35,4 +35,51 @@ protected double calculateStdDev(IntArrayList degrees, double avg) {
         }
         return Math.sqrt(sum / degrees.size());
     }
+    
+    /**
+     * The method calculates the StdDev of an array of updated degrees.
+     * @param triple - edge on which graph operation is performed.
+     * @param graph - input graph.
+     * @param graphOperation - boolean value indicating graph operation. ("true" for adding an edge and "false" for removing an edge)
+     * @param previousResult - UpdatableMetricResult object containing the previous computed results.
+     * @return SimpleMetricResult object.
+     */
+    @Override
+    public SimpleMetricResult update(TripleBaseSingleID triple, ColouredGraph graph,
+            boolean graphOperation, UpdatableMetricResult previousResult, VertexDegrees mVertexDegrees) {
+    	SimpleMetricResult newMetricResult = (SimpleMetricResult) previousResult;
+    	int[] degreesArray;
+    	IntArrayList degreesList;
+		switch (getName()) {
+			case "stdDevInDegree":	
+				degreesArray = mVertexDegrees.getmMapVerticesinDegree();
+				degreesList =  new IntArrayList(degreesArray);// Arrays.asList(ArrayUtils.toObject(degreesArray));
+				newMetricResult.setResult(calculateStdDev(degreesList, calculateAvg(degreesList)));
+				break;
+
+			case "stdDevOutDegree":
+				degreesArray = mVertexDegrees.getmMapVerticesoutDegree();
+				degreesList = new IntArrayList(degreesArray);
+				newMetricResult.setResult(calculateStdDev(degreesList, calculateAvg(degreesList)));
+				break;
+			
+			default:// If metric is other than maxInDegree and maxOutDegree then apply the metric
+				newMetricResult = applyUpdatable(graph, graphOperation, triple, previousResult);
+		} 
+		
+        return newMetricResult;
+    }
+    
+    /** Stores the previously computed values in SimpleMetricResult object.
+     * @param graph - input graph.
+     * @param graphOperation - boolean value indicating graph operation. ("true" for adding an edge and "false" for removing an edge). Not used here.
+     * @param triple - Edge on which graph operation is performed. Not used here.
+     * @param newMetricResult - UpdatableMetricResult object containing the results that should be updated.
+     * @return SimpleMetricResult object.
+     */
+    @Override
+    public SimpleMetricResult applyUpdatable(ColouredGraph graph, boolean graphOperation, TripleBaseSingleID triple, UpdatableMetricResult newMetricResult) {
+    	SimpleMetricResult metricResultTempObj = (SimpleMetricResult) newMetricResult;
+    	return metricResultTempObj;

```

Comment:
This implementation is wrong. The given, previous result is simply returned. Please have a look at the default implementation in the interface.

Reply:
The idea behind just passing the previous result in applyUpdatable is to avoid the apply method from being called again.

In the EdgeModifier class, we already calculate the metric value in tryToAddAnEdge/tryToRemoveAnEdge method, 
we thought to reuse it again while calling executeAddingAnEgde/execteRemovingAnEdge.

We are thinking about a better way to implement this.